Sources/JSONRequest.swift

```diff
@@ -77,15 +77,19 @@ open class JSONRequest {
     open static var requestTimeout = 5.0
     open static var resourceTimeout = 10.0
     open static var requestCachePolicy: NSURLRequest.CachePolicy = .useProtocolCachePolicy
-    open static var urlSession: URLSession?
 
     open var httpRequest: NSMutableURLRequest? {
         return request
     }
 
-    internal static var requireNetwork = true
+    /* Used for dependency injection of outside URLSessions (keep nil to use default) */
+    private var urlSession: URLSession?
 
-    public init() {
+    /* Set to false during testing to avoid test failures due to lack of internet access */
+    internal static var requireNetworkAccess = true
+
+    public init(session: URLSession? = nil) {
+        urlSession = session
```

Comment:
Can you add a comment explaining the behavior when you omit the `session` parameter? Then, Good to go.  I'll mark as approved.